### PR TITLE
Theme setting read from db and write to db

### PR DESCRIPTION
**Pull request by fuzheng1998**
write back to DB when theme change

### CODE REVIEW

**Review by ThuvaMathu — COMMENTED** [review]
Lgtm. Let's see in action.

**Comment by fuzheng1998** [comment]
fetching theme from DB works now.  @kovarny

File: lib/state/theme_state_notifier.dart [hunk 1]
```diff
@@ -1,12 +1,44 @@
...
+  ThemeStateNotifier(bool loginState) : super(false) {
+    if (loginState == true && auth.currentUser != null) {
+      var userSettingDoc =
+          dbInstance.collection('user').doc(auth.currentUser!.uid);
+      Future theme = userSettingDoc.get().then((value) {
+        print('theme' + value.toString());
+        return value['themeMode'];
+      });
+      theme
+          .then((value) {
```
Comment (kovarny) [inline lib/state/theme_state_notifier.dart:20]: why chaining if it is just one action to assign the state?

File: lib/state/theme_state_notifier.dart [hunk 2]
```diff
@@ -1,12 +1,44 @@
...
+            bool isDark = value == 'light' ? false : true;
```
Comment (kovarny) [inline lib/state/theme_state_notifier.dart:21]: Don't introduce variables for no reason.

File: lib/state/theme_state_notifier.dart [hunk 3]
```diff
@@ -1,12 +1,44 @@
...
+      // print(userSettingDoc);
+    }
   }
 }
 
 final themeStateNotifierProvider =
-    StateNotifierProvider<ThemeStateNotifier, bool>(
-        (ref) => ThemeStateNotifier(false));
+    StateNotifierProvider<ThemeStateNotifier, bool>((ref) {
+  var loginState = ref.watch(isLoggedIn);
```
Comment (kovarny) [inline lib/state/theme_state_notifier.dart:42]: Same

File: lib/state/theme_state_notifier.dart [hunk 4]
```diff
@@ -1,12 +1,44 @@
...
+          })
+          .then((isDark) => state = isDark)
+          .whenComplete(() => print(state));
+    }
+  }
   void changeTheme() {
     state = !state;
+    String themeMode = state == false ? 'light' : 'dark';
+    if (auth.currentUser != null) {
+      var userSettingDoc =
```
Comment (kovarny) [inline lib/state/theme_state_notifier.dart:32]: unneeded variable

**Review by kovarny — CHANGES_REQUESTED** [review]

**Review by kovarny — APPROVED** [review]

**Pull request merged**